tests/Benchmark/KVOBenchmarkTests.mm

```diff
@@ -0,0 +1,147 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <Starboard/SmartTypes.h>
+#import "Benchmark.h"
+#import <CppUtils.h>
+#import <atomic>
+
+// Minimal object that supports KVO notifications for a property
+@interface TestKVOObject : NSObject
+@property NSInteger intProperty;
+@end
+
+@implementation TestKVOObject
+@end
+
+// Minimal object for observing to TestKVOObject
+static std::atomic<size_t> s_observedCount;
+@interface TestKVOObserver : NSObject
+@end
+
+@implementation TestKVOObserver
+- (void)observeValueForKeyPath:(NSString*)keyPath ofObject:(id)object change:(NSDictionary*)change context:(void*)context {
+    ++s_observedCount;
+}
+@end
+
+class AddObservers : public ::benchmark::BenchmarkCaseBase {
+    StrongId<TestKVOObject> _testKVOObject;
+    StrongId<NSMutableArray> _testObservers;
+
+public:
+    void PreRun() {
+        _testKVOObject.attach([TestKVOObject new]);
+        _testObservers.attach([NSMutableArray new]);
+        @autoreleasepool {
+            for (size_t i = 0; i < 1000; ++i) {
+                [_testObservers addObject:[[NSObject new] autorelease]];
+            }
+        }
+    }
+
+    inline void Run() {
+        for (NSObject* obj in _testObservers.get()) {
+            [_testKVOObject addObserver:obj forKeyPath:@"intProperty" options:0 context:nullptr];
```

Comment:
I think it would end up being harder to read because of the need to construct an NSIndexSet. Leaving as-is.